### ISSUE TITLE
Deprecated Gradle features were used, making it incompatible with Gradle 5.0

### ISSUE THREAD

**Issue by vorburger** [tbroyer/gradle-errorprone-plugin-v0.0.x#66]
Thank you for making this Gradle plugin! [In a project where I'm using this](https://github.com/vorburger/minecraft-storeys-maker/blob/master/build.gradle), Gradle (v4.6) says:

    Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.```

and [when run with](https://docs.gradle.org/4.6/userguide/command_line_interface.html#sec:command_line_warnings) `./gradlew build --warning-mode=all` reveals:

`Putting annotation processors on the compile classpath has been deprecated and is scheduled to be removed in Gradle 5.0. Please add them to the processor path instead. If these processors were unintentionally leaked on the compile classpath, use the -proc:none compiler option to ignore them..
`

I gather this is probably because Gradle doesn't think that the `error_prone_core.JAR` which contains a `/META-INF/services/com.sun.source.util.Plugin` (for `com.google.errorprone.ErrorProneJavacPlugin`), should be on the main compile classpath? I guess it's right, in an ideal world, only the error_prone_annotations.jar (but NOT the error_prone_annotation.jar, note the *s) published by the Error Prone project should be on the compile classpath, and the "rest" (error_prone_core.JAR ?) should be... I'm not sure, on the annotation processor path, somehow, if that is or could be made possible with this Gradle plugin?

**Comment by tbroyer** [comment]
I highly doubt this is due to either this plugin or error prone. Does the warning really disappear when you stop using the plugin‽

> I gather this is probably because Gradle doesn't think that the error_prone_core.JAR which contains a `/META-INF/services/com.sun.source.util.Plugin` (for `com.google.errorprone.ErrorProneJavacPlugin`), should be on the main compile classpath?

Error prone is not in the compile classpath (well, in your project, it is; but it shouldn't), and Gradle doesn't care about `com.sun.source.util.Plugin` in its annotation processors detection; so this couldn't be the cause.

At first sight, I'd say this is caused by the SpongeAPI, which includes annotation processors. Either add it to the `annotationProcessor` configuration as well (if you need the processors), or pass `-proc:none` as a `compilerArgs` to your `JavaCompile` tasks (`tasks.withType(JavaCompile) { options.compilerArgs << '-proc:none' }`)

**Comment by vorburger** [comment]
> At first sight, I'd say this is caused by the SpongeAPI, which includes annotation processors.

duh! So sorry; you are absolutely right, I got totally confused here; it's not caused by errorprone.

> Either add it to the annotationProcessor configuration as well (if you need the processors),
> or pass -proc:none as a compilerArgs to your JavaCompile tasks

Thanks a lot for these helpful tips! On it in https://github.com/vorburger/minecraft-storeys-maker/issues/22 ..

> Error prone is not in the compile classpath (well, in your project, it is; but it shouldn't),

Oh, now I get it, sorry; so one would typically want to use something this:

    dependencies {
      errorprone 'com.google.errorprone:error_prone_core:2.2.0'
      implementation 'com.google.errorprone:error_prone_annotations:2.2.0'
    }

instead of this:

    dependencies {
      implementation 'com.google.errorprone:error_prone_core:2.2.0'
    }

[right - got it](https://github.com/vorburger/minecraft-storeys-maker/pull/24). Let me know if you think a PR to your README clarifying this would add any value.

Sorry again for the noise.

**Comment by tbroyer** [comment]
> Let me know if you think a PR to your README clarifying this would add any value.

Well, the README only talks about the `errorprone` configuration. Whether you use error prone annotations is up to you (I rarely do), and then they indeed need to go into your compile classpath (versions are not required to match, I inherit those transitively from Guava for example without any problem); but that's no different from other annotations that error prone will use (checkerframework, jsr350, Dagger, `@Beta`, etc.
TL;DR: I don't think it's worth adding it to the README. (it's the first time I hear feedback of that sort; I'll of course reevaluate if I receive others)

**Comment by CoolSuplex** [comment]
gradle 4.8 is also giving me this warning ---
Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.

**Comment by tbroyer** [comment]
@CoolSuplex Please rerun with `--warning-mode=all`. I'm 100% positive that this is unrelated to this plugin (as I'm using it with Gradle 4.8 without this warning)

**Comment by CoolSuplex** [comment]
Thanks @tbroyer I passed that but am facing this new one now,

Synchronize gradle projects with workspace failed due to an unexpected error
Java Model Exception: Java Model Status [Cannot nest output folder ’some-sample-app/bin/main' inside output folder 'some-sample-app/bin']

**Comment by yichunzhao** [comment]
I meet this error as using the gradle 4.7 too; so what is the solution??

**Comment by Tonterias** [comment]
Same error using Gradle v4.6

Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.

I try to upgrade and it does not let me do it

gradlew wrapper --gradle-version=4.8.1 --distribution-type=bin

**Comment by tbroyer** [comment]
# To anyone seeing that warning

Please, do as suggested in the warning and re-run your build with `--warning-mode=all` to see what the deprecated feature is; it might have nothing to do with this plugin (use `--rerun-tasks` to make sure you'll meet the warning again; and if you enabled build cache, don't forget `--no-build-cache` too)

And of course try building your project without Error Prone (comment out the line that applies the plugin, the line that configures the dependency, and possibly the Error Prone-specific `options.compilerArgs`) to see if you also have that warning (again, don't forget `--rerun-tasks` and `--no-build-cache`)

**Comment by tbroyer** [comment]
> I try to upgrade and it does not let me do it
> 
> gradlew wrapper --gradle-version=4.8.1 --distribution-type=bin

Fwiw, I generally go edit the `gradle/wrapper/gradle-wrapper.properties file directly, then run `./gradlew wrapper` to update the `gradlew*` scripts and `gradle/wrapper/gradle-wrapper.jar`.

**Comment by reiva5** [comment]
I've tried it using flag `--rerun-tasks` and `--no-build-cache` (also using one of them separately, and of course using flag `--warning-mode=all` too) still not worked for me. Besides that, there is addition `error` like this `> Could not find method processor() for arguments [...] on object type of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler` besides `error` like this thread said. Is this issue still related to this thread?

**Comment by AllenVork** [comment]
I'm using gradle 4.9 and get the same error when executing "gradle artifactoryPublish":
```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':pcore:artifactoryPublish'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:51)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:46)
        at org.gradle.execution.taskgraph.LocalTaskInfoExecutor.execute(LocalTaskInfoExecutor.java:42)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:273)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareWorkItemExecutor.execute(DefaultTaskExecutionGraph.java:258)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:135)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:130)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.execute(DefaultTaskPlanExecutor.java:200)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.executeWithWork(DefaultTaskPlanExecutor.java:191)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: java.lang.NoSuchMethodError: org.gradle.api.publish.maven.internal.publisher.MavenProjectIdentity.getArtifactId()Ljava/lang/String;
        at org.jfrog.gradle.plugin.artifactory.task.helper.TaskHelperPublications.getArtifactDeployDetails(TaskHelperPublications.java:216)
        at org.jfrog.gradle.plugin.artifactory.task.helper.TaskHelperPublications.collectDescriptorsAndArtifactsForUpload(TaskHelperPublications.java:147)
        at org.jfrog.gradle.plugin.artifactory.task.ArtifactoryTask.collectDescriptorsAndArtifactsForUpload(ArtifactoryTask.java:70)
        at org.jfrog.gradle.plugin.artifactory.task.BuildInfoBaseTask.prepareAndDeploy(BuildInfoBaseTask.java:367)
        at org.jfrog.gradle.plugin.artifactory.task.BuildInfoBaseTask.collectProjectBuildInfo(BuildInfoBaseTask.java:129)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:786)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:753)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
        ... 31 more
```

**Comment by jbduncan** [comment]
Hi @AllenVork, I don't know if this will help, but I'd suggest you try @tbroyer's suggestions in his [earlier comment](https://github.com/tbroyer/gradle-errorprone-plugin/issues/66#issuecomment-402662180) to check if the error you're experiencing is indeed gradle-errorprone-plugin's fault.

I think that should in turn allow us to help you understand why things aren't working for you. :)

**Comment by tbroyer** [comment]
@AllenVork Nothing to do with this plugin. Artifactory plugin seems to make use of internal Gradle classes, that might have been updated in 4.9; last version of that plugin was 2 weeks ago, so it's likely that it's simply not compatible with Gradle 4.9 (IIRC there was already a regression with 4.8, probably fixed by that 4.7.5 release). So make sure you're using the latest version of `org.jfrog.artifactory`, and if that doesn't fix the problem, then :man_shrugging:

**Comment by tbroyer** [comment]
I'm afraid I have to lock this issue, as everybody commenting here seems to have found this issue from a Google/GitHub search and didn't take the time to understand the warnings or errors they were seeing and whether they were actually related to this project.

If you're coming here and find this conversation locked, please follow the instructions in https://github.com/tbroyer/gradle-errorprone-plugin/issues/66#issuecomment-402662180 and if you can confirm it is this plugin that generates a deprecation warning, then please file another issue.
